Adafruit_PN532.cpp

```diff
@@ -1652,8 +1652,8 @@ void Adafruit_PN532::writecommand(uint8_t* cmd, uint8_t cmdlen) {
     delay(2);     // or whatever the delay is for waking up the board
     spi_write(PN532_SPI_DATAWRITE);
 
-    checksum = PN532_PREAMBLE + PN532_PREAMBLE + PN532_STARTCODE2;
-    spi_write(PN532_PREAMBLE);
+    checksum = PN532_PREAMBLE + PN532_STARTCODE1 + PN532_STARTCODE2;
+    spi_write(PN532_STARTCODE1);
     spi_write(PN532_PREAMBLE);
```

Comment:
```
    spi_write(PN532_STARTCODE1);
    spi_write(PN532_PREAMBLE);
```

should be:

```
    spi_write(PN532_PREAMBLE);
    spi_write(PN532_STARTCODE1);
```

Right?

Reply:
@armadillojim, I think so, yes. Preamble should always come first.